### PR TITLE
[ENG-1644] bug overwrite cassette

### PR DESCRIPTION
**Pull request by yardz**
My last changes introduced a bug. When running the tests (using the VCR) I was erasing the existing cassettes by "[]".
This PR is fixing this bug. Now only when not listening to cassette for a test will put a "[]"

### CODE REVIEW

File: src/VCR.ts [hunk 1]
```diff
@@ -62,10 +62,13 @@ async function writeCassetteFile(cassettes: CassetteFile): Promise<void> {
 async function saveCassettes(defns: nock.Definition[]): Promise<void> {
   const cassettes: CassetteFile = await readCassetteFile()
   const testName = expect.getState().currentTestName
-  if (testName) {
-    cassettes[testName] = defns
-    await writeCassetteFile(cassettes)
-  }
+  if (!testName) return
```
Comment (WarrickFitz) [inline src/VCR.ts:66]: Under what circumstance would there be no test name?

File: src/VCR.ts [hunk 2]
```diff
@@ -62,10 +62,13 @@ async function writeCassetteFile(cassettes: CassetteFile): Promise<void> {
...
+
+  const hasCasset = cassettes[testName] && !!cassettes[testName].length
```
Comment (WarrickFitz) [inline src/VCR.ts]: This is pretty hard to follow can you add a comment please.

File: src/VCR.ts (hunk 1 above)
Comment (yardz) [reply src/VCR.ts:66]: I don't know. That condition was there, so I didn't change it.
I'm guessing it's to prevent someone from accidentally putting this into the normal code

Comment (WarrickFitz) [reply src/VCR.ts:66]: ok, wil resolve.

File: src/VCR.ts (hunk 2 above)
Comment (yardz) [reply src/VCR.ts]: I changed the code and made it more readable

**Review by WarrickFitz — APPROVED** [review]

**Review by fractallian — APPROVED** [review]
